src/compiler/commandLineParser.ts

```diff
@@ -1034,6 +1037,7 @@ const commandOptionsWithoutBuild: CommandLineOption[] = [
         name: "paths",
         type: "object",
         affectsModuleResolution: true,
+        allowConfigDirTemplateSubstitution: true,

```

Comment:
Long term, I do wonder if we'll need some other flag here given we'll introduce more substitutions, but thankfully this is internal!